### DISCUSSION TITLE
ShapeSys question

### DISCUSSION THREAD

**Discussion by pmalecki0** [scikit-hep/pyhf#2347] [Q&A]
Dear Experts!

I am fitting a model in which one of the samples has an bin-by-bin uncorrelated uncertainty (which is not statistical uncertainty, but a systematic one derived from external package). For this I am using ShapeSys - I hope it is the right thing to do.
My question is - how should I define the contents of the corresponding ShapeSys histogram? 
Since the output shows "gamma_* " nuisance parameters corresponding to this ShapeSys, I believe these are multiplicative.
So, should the content of the given ShapeSys bin be relative error? I.e. if the bin content is 0 then there is no uncertainty and if it is 1 then it is 100%? 

Best regards,
Paweł

**Comment by alexander-held** [comment]
Hi Paweł, `shapesys` indeed implements the functionality you are looking for. It works like `staterror` when it comes to specifying the `"data"` field: you provide the absolute uncertainties. See also the relevant documentation here: https://pyhf.readthedocs.io/en/v0.7.4/likelihood.html#uncorrelated-shape-shapesys.

What is multiplicative about this is that the value of the nuisance parameter multiplies the event yield in the respective bin, so for a bin with nominal prediction of 10 events and a `shapesys` nuisance parameter fitted as 1.1, you would have a post-fit yield of 11 events. The naming of the parameter is up to you: it will be called `"gamma_*"` only if you give it that name. That prefix is a common one though, given that `shapsys` parameters use a Poisson constraint term.

**Reply by marfari** [reply]
I am building error histograms for the shapesys modifier and I am writing xml files which include this line:

`<ShapeSys Activate="True" Name="shapesys_sig" HistoPath="Signal" HistoName="error" ConstraintType="Poisson" />`

I then use the `xml2json` command to convert the xml files into a .json workspace for pyhf

When I do this I noticed that I need to build the histograms with the relative errors in order for the data field of shapesys in the .json file to contain the absolute uncertainties

So it seems that in the conversion, it multiplies the entries of my error histograms by the entries of my nominal histograms to build the data field of shapesys

Is this the correct implementation? I.e. should I indeed build the error histograms with the relative errors?
